QuizApp/App/Networking/User/UserNetworkClient.swift

```diff
@@ -0,0 +1,51 @@
+import Foundation
+
+protocol UserNetworkClientProtocol {
+
+    var userInfo: UserResponseModel? { get async throws }
+
+    func update(name: String) async throws
+}
+
+final class UserNetworkClient: UserNetworkClientProtocol {
+
+    private let networkClient: NetworkClientProtocol
+    private let secureStorage: SecureStorageProtocol
+
+    init(networkClient: NetworkClientProtocol, secureStorage: SecureStorageProtocol) {
+        self.networkClient = networkClient
+        self.secureStorage = secureStorage
+    }
+
+    var userInfo: UserResponseModel? {
+        get async throws {
+            guard let url = URL(string: Endpoint(type: .account).path) else {
```

Comment:
If URL cannot be created, that is an Error and you should throw it. It is the developer's fault that he is trying to use non buildable URL.